pkg/controller/worker/machines.go

```diff
@@ -202,7 +202,7 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 			}
 
 			if pool.NodeTemplate != nil {
-				var zoneName string
+				zoneName := "no-zone"
```

Comment:
Wouldn’t the more sustainable approach be to make the field optional and simply omit it when the azure Shoot is not zoned one?

Reply:
Yes I though abt that. The only issue is it would require MCM api change as `NodeTemplate` struct is there, and then changes in CA are required for validation, and revendoring new MCM release in all the extensions, which will delay scale-from-zero even further.
Since this is just Azure which faces this problem, so looking at the cost to benefit ratio, I took this approach.

Reply:
Okay, I am fine with this change as a short-term mitigation. But in general it does not make sense to set dummy values for fields when they are not used but wrongly required in the CRD. In long-term I would suggest to open an issue about it, add a TODO here in provider-azure, and then do the required changes to make this field optional. WDYT?

Reply:
Yes definitely, long term we need to make this optional. I'll do the required changes.

Reply:
Let's go with @ialidzhikov suggestion. Can you maybe add a comment/TODO to explain why we use the no-zone string and that it should be updated in the future ? Otherwise /lgtm

Reply:
updated, I'll raise an issue in g/g now